app/services/targets/create_service.rb

```diff
@@ -34,10 +34,15 @@ def create_match(target, compatible_target)
       [target, compatible_target].each do |target_to_update|
         target_to_update.update!(matched: true)
       end
+      notification(target.user)
     end
 
     def compatible_target(target)
       TargetQuery.new.compatible_target(target)
     end
+
+    def notification(user)
+      NotificationService.send_match_email(user).deliver
```

Comment:
Is this actually working? Looks like you are calling `send_match_email` that is defined in the mailer, not in the service class